src/couch_index/src/couch_index_compactor.erl

```diff
@@ -81,6 +85,14 @@ handle_cast(_Mesg, State) ->
 
 handle_info({'EXIT', Pid, normal}, #st{pid=Pid}=State) ->
     {noreply, State#st{pid=undefined}};
+handle_info({'EXIT', Pid, Reason}, #st{pid = Pid} = State) ->
+    #st{idx = Idx, mod = Mod} = State,
+    {ok, IdxState} = gen_server:call(Idx, {compaction_failed, Reason}),
```

Comment:
I think the following would be better. Since it would allow us to hide the internal representation. 
`{ok, IdxState} = couch_index:handle_failure(Pid, Reason)`